core/io/json.cpp

```diff
@@ -588,10 +588,741 @@ void JSON::_bind_methods() {
+					ERR_FAIL_COND_V(!d.has("size"), Variant());
+					AABB r;
+					r.position = to_native(d["position"]);
+					r.size = to_native(d["size"]);
+					return r;
+				} else if (type == Variant::get_type_name(Variant::BASIS)) {
+					ERR_FAIL_COND_V(!d.has("x"), Variant());
+					ERR_FAIL_COND_V(!d.has("y"), Variant());
+					ERR_FAIL_COND_V(!d.has("z"), Variant());
+					Basis b;
+					b.set_column(0, to_native(d["x"]));
+					b.set_column(1, to_native(d["y"]));
+					b.set_column(2, to_native(d["z"]));
+					return b;
+				} else if (type == Variant::get_type_name(Variant::TRANSFORM3D)) {
+					ERR_FAIL_COND_V(!d.has("basis"), Variant());
+					ERR_FAIL_COND_V(!d.has("origin"), Variant());
+					Transform3D t;
+					t.basis = to_native(d["basis"]);
+					t.origin = to_native(d["origin"]);
+					return t;
+				} else if (type == Variant::get_type_name(Variant::PROJECTION)) {
+					ERR_FAIL_COND_V(!d.has("x"), Variant());
+					ERR_FAIL_COND_V(!d.has("y"), Variant());
+					ERR_FAIL_COND_V(!d.has("z"), Variant());
+					ERR_FAIL_COND_V(!d.has("w"), Variant());
+					Projection p;
+					p[0] = to_native(d["x"]);
+					p[1] = to_native(d["y"]);
+					p[2] = to_native(d["z"]);
+					p[3] = to_native(d["w"]);
+					return p;
+				} else if (type == Variant::get_type_name(Variant::COLOR)) {
+					// TODO
+				} else if (type == Variant::get_type_name(Variant::NODE_PATH)) {
+					ERR_FAIL_COND_V(!d.has("path"), Variant());
+					NodePath np = d["path"];
+					return np;
+				} else if (type == Variant::get_type_name(Variant::STRING_NAME)) {
+					ERR_FAIL_COND_V(!d.has("name"), Variant());
+					StringName s = d["name"];
+					return s;
+				} else if (type == Variant::get_type_name(Variant::OBJECT)) {
+					ERR_FAIL_COND_V(!d.has("type"), Variant());
+					ERR_FAIL_COND_V(!d.has("properties"), Variant());
+
+					ERR_FAIL_COND_V(!p_allow_classes, Variant());
+
+					String type = d["type"];
+					bool is_script = type == "Script" || ClassDB::is_parent_class(type, "Script");
+					ERR_FAIL_COND_V(!p_allow_scripts && is_script, Variant());
+					Object *obj = ClassDB::instantiate(type);
+					ERR_FAIL_COND_V(!obj, Variant());
+
+					Dictionary p = d["properties"];
+
+					List<Variant> keys;
+					p.get_key_list(&keys);
+
+					for (const Variant &K : keys) {
+						String property = K;
+						Variant value = to_native(p[K], PASS_ARG);
+						obj->set(property, value);
+					}
+
+					Variant v(obj);
+
+					return v;
+				} else if (type == Variant::get_type_name(Variant::DICTIONARY)) {
+					ERR_FAIL_COND_V(!d.has("pairs"), Variant());
+					Array pairs = d["pairs"];
+					Dictionary r;
+					for (int i = 0; i < pairs.size(); i++) {
+						Dictionary p = pairs[i];
+						ERR_CONTINUE(!p.has("key"));
+						ERR_CONTINUE(!p.has("value"));
+						r[to_native(p["key"], PASS_ARG)] = to_native(p["value"]);
+					}
+					return r;
+				} else if (type == Variant::get_type_name(Variant::ARRAY)) {
+					// TODO
```

Comment:
Dictionary was there twice with the same code, seems like you forgot to edit it after copying to handle Array.

Reply:
Fixed, Array doesn't use GDTYPE so it doesn't need to be handled here (it's handled below).
I added an error check in case this changes, but we could also just remove this branch.